### PR TITLE
docs: updated description for missing samplers and pruners

### LINKED ISSUE #3072: Add missing sampler and pruner classes to a tutorial page
[Efficient Optimization Algorithms](https://optuna.readthedocs.io/en/latest/tutorial/10_key_features/003_efficient_optimization_algorithms.html) tutorial provides lists of samplers/pruners implemented in Optuna. However a few classes are not on the list. It would be great to mention them comprehensively from the tutorial pages for Optuna users. 

More concretely, please add the following entries to the tutorial page.

## Samplers

- `optuna.samplers.PartialFixedSampler`
- `optuna.samplers.NSGAIISampler`

## Pruners
- `optuna.pruners.PatientPruner`
- `optuna.pruners.PercentilePruner`

### PR DESCRIPTION
**Pull request by masaaldosey**
## Motivation

This PR fixes #3072 and aims at providing comprehensive lists of samplers/pruners implemented in Optuna for the users. 

## Description of the changes

Updated the description to include the missing :
- samplers: `PartialFixedSampler` and `NSGAIISampler`
- pruners: `PatientPruner` and `PercentilePruner`. 

Further, the description has been updated such that the sampler/pruner class names are in alphabetical order.

### CODE REVIEW

**Comment by nzw0301** [comment]
Great! Could you apply `black`, ideally, [`formats.sh`](https://github.com/optuna/optuna/blob/master/formats.sh)?

**Comment by l-e-e-l-a** [comment]
> Great! Could you apply `black`, ideally, by using [`formats.sh`](https://github.com/optuna/optuna/blob/master/formats.sh)?

i went through the contributing guide and ran the `./formats.sh` command.
image

**Comment by nzw0301** [comment]
Interesting. maybe due to the difference of libraries version or OS. I try to give a concrete modification to this PR.

File: tutorial/10_key_features/003_efficient_optimization_algorithms.py [hunk 1]
```diff
@@ -60,12 +64,18 @@
...
 # - Hyperband algorithm implemented in :class:`optuna.pruners.HyperbandPruner`
 #
 # - Median pruning algorithm implemented in :class:`optuna.pruners.MedianPruner`
 #
+# - Non-pruning algorithm implementd in :class:`optuna.pruners.NopPruner`
+#
+# - Algorithm to operate pruner with tolerance implemented in :class:`optuna.pruners.PatientPruner`
+#
+# - Algorithm to prune specified number of trials implemented in :class:`optuna.pruners.PercentilePruner`
+#
```
Comment (nzw0301) [inline tutorial/10_key_features/003_efficient_optimization_algorithms.py:76]:
```suggestion
#
```

Comment (nzw0301) [reply tutorial/10_key_features/003_efficient_optimization_algorithms.py:76]: Could you remove this space?

**Comment by toshihikoyanase** [comment]
@nzw0301 Could you continue the review, please? I think you can merge this PR with a single approval since it only updates the document.

**Comment by nzw0301** [comment]
## Ordering 

How about following the ordering on the documentation rather than alphabetical ordering?
I mean the orders on the following pages:
- https://optuna.readthedocs.io/en/latest/reference/pruners.html
- https://optuna.readthedocs.io/en/latest/reference/samplers.html

File: tutorial/10_key_features/003_efficient_optimization_algorithms.py [hunk 2]
```diff
@@ -17,12 +17,16 @@
...
 
-- Tree-structured Parzen Estimator algorithm implemented in :class:`optuna.samplers.TPESampler`
-
 - CMA-ES based algorithm implemented in :class:`optuna.samplers.CmaEsSampler`
 
 - Grid Search implemented in :class:`optuna.samplers.GridSampler`
 
+- Nondominated Sorting Genetic Algorithm II implemented in :class:`optuna.samplers.NSGAIISampler`
+
+- Algorithm to enable Partial Fixed parameters implemented in :class:`optuna.samplers.PartialFixedSampler`
```
Comment (nzw0301) [inline tutorial/10_key_features/003_efficient_optimization_algorithms.py:26]:
```suggestion
- Algorithm to enable partial fixed parameters implemented in :class:`optuna.samplers.PartialFixedSampler`
```

File: tutorial/10_key_features/003_efficient_optimization_algorithms.py [hunk 3]
```diff
@@ -60,12 +64,18 @@
...
-#
 # - Hyperband algorithm implemented in :class:`optuna.pruners.HyperbandPruner`
 #
 # - Median pruning algorithm implemented in :class:`optuna.pruners.MedianPruner`
 #
+# - Non-pruning algorithm implementd in :class:`optuna.pruners.NopPruner`
+#
+# - Algorithm to operate pruner with tolerance implemented in :class:`optuna.pruners.PatientPruner`
+#
+# - Algorithm to prune specified number of trials implemented in :class:`optuna.pruners.PercentilePruner`
```
Comment (nzw0301) [inline tutorial/10_key_features/003_efficient_optimization_algorithms.py:75]: How about using the documentation's words:
```suggestion
# - Algorithm to prune specified percentile of trials implemented in :class:`optuna.pruners.PercentilePruner`
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]
I've left a few comments. What do you think about it?

**Comment by masaaldosey** [comment]
> I've left a few comments. What do you think about it?

have made suggested changes :)

**Review by nzw0301 — APPROVED** [review]
Great! Thank you for your contributions!

**Pull request merged**